### PR TITLE
Add recipe for docstring_parser

### PR DESCRIPTION
**Pull request by Chilipp**
This PR adds the recipe for the docstring-parser package based on the corresponding [project on pypi.io](https://pypi.org/project/docstring-parser/).

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

@rr-, please quickly confirm in this PR that you are willing to be listed as a maintainer here.

closes https://github.com/rr-/docstring_parser/issues/23

### CODE REVIEW

File: recipes/docstring-parser/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  summary: 'Parse Python docstrings in reST and Google format '
+  description: |
+    Parse Python docstrings. Currently support ReST, Google, and Numpydoc-style
+    docstrings.
+  dev_url: https://github.com/rr-/docstring_parser
+
+extra:
+  recipe-maintainers:
+    - Chilipp
+    - rr-
```
Comment (scopatz) [inline recipes/docstring-parser/meta.yaml:42]: @rr- are you willing to be a maintainer?

**Comment by Chilipp** [comment]
sorry, it's `docstring_parser`, not `docstring-parser`

File: recipes/docstring-parser/meta.yaml (hunk 1 above)
Comment (rr-) [reply recipes/docstring-parser/meta.yaml]: Yes

**Comment by Chilipp** [comment]
Alright, thanks @rr-! @conda-forge/help-python, from the contributors perspective, this is ready to be merged

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**